crates/amaru/src/bin/amaru/cmd/import_ledger_state.rs

```diff
@@ -285,8 +405,22 @@ fn decode_new_epoch_state(
     d.skip()?;
     // Constitution
     d.skip()?;
+
+    info!(what = "pparams");
+
     // Current Protocol Params
-    let protocol_parameters = import_protocol_parameters(db, &epoch, d.decode()?)?;
+    let pparams = if let Some(pparams_dir) = pparams_dir {
+        let pparams_hash: &minicbor::bytes::ByteSlice = d.decode()?;
+        let pparams_file_path = fs::read_dir(pparams_dir)?
+            .filter_map(|entry| entry.ok().map(|e| e.path()))
+            .find(|path| path.file_name().map(|filename| filename.to_str() == Some(&hex::encode(pparams_hash.as_ref()))).unwrap_or(false))
+            .ok_or(Error::MissingPparamsFile)?;
+        let pparams_file = fs::read(pparams_file_path)?;
+        cbor::Decoder::new(&pparams_file).decode()?
+    } else {
+        d.decode()?
+    };
```

Comment:
I don't understand why this is necessary? If the goal is to test against a NES, why not use the current protocol params specified in that NES ?